docs/getting-started.md

```diff
@@ -3,126 +3,220 @@ id: getting-started
+
+6. **Watcher Attestation**: Once a winning bid is selected, the watcher attests it on the Fast [Switchboard](/switchboards) on the target chain.
+
+7. **Onchain Execution**: The [Transmitter](/transmitters) executes the deployment on the target chain through the SOCKET contract, which verifies with the [Switchboard](/switchboards) that the Transmitter is authorized to execute the payload.
+
+8. **Event Observation**: The [Watcher](/watchers) observes the execution events and resolves the `AsyncPromise` with the deployment result.
+
+9. **Callback Execution**: This triggers the callback in the `AppGateway`, which:
+   - Stores the deployed contract address
+   - Deploys a Forwarder contract (an EVMx representation of the onchain deployed contract)
+   - Sets up the Forwarder to handle future multi-chain calls
+
+:::note Note
+The Fast Switchboard used here is just one example of a simplified [Switchboard](/switchboards) implementation that allows a trusted [Watcher](/watchers) to attest that a given [Transmitter](/transmitters) is authorized to execute a specific payload. Switchboards can be customized to handle various verification methods and proof systems depending on your security requirements.
+:::
+
+<div style={{ display: 'flex', justifyContent: 'center' }}>
+    <img src="/img/deploy_flow.png" alt="switchboard diagram" style={{ width: '100%' }} />
+</div>
+
+
+### Verify transaction completion
+
+Chain-abstracted deployments aren't synchronous as shown in the flow above. You queue them within the app gateway and they get picked up by a transmitter to be relayed onchain. Let's verify that our `Counter` instances have been successfully deployed on all chains:
+
+```bash
+# Replace <TX_HASH> with the transaction hash from your deployment output for each chain:
+curl https://api-evmx-devnet.socket.tech/getDetailsByTxHash?txHash=<TX_HASH>
+```
+
+Ensure the status is `COMPLETED`. For convenience, you can monitor all deployment transactions at once using:
+
+```bash
+node script/transactionStatus.js DeployOnchainCounters
+```
+
+:::tip
+The deployment process may take a few minutes as transactions need to confirm on each target chain.
+:::
+
+## Increment multiple counters
+
+Now that the Counter instances have been deployed we can interact with them through a single interface, our `CounterAppGateway`. We will increment the counter on each chain by calling `incrementCounters`:
+
+```bash
+forge script script/counter/IncrementCountersFromApp.s.sol --broadcast --skip-simulation --legacy --with-gas-price 0
+```
+
+Behind the scenes this works similarly to the deployment process. The only difference is that we queue our onchain call to the `Counter` instance via their EVMx representation, the [forwarder](/forwarder-addresses) contract that was registered during deployment. Also, this time no callback is registered to run for when the counter has been successfully incremented on chain.
+
+### Monitoring Transaction Status
```

Comment:
Should we remove this monitoring section as we already showed above how to do it? Most transactions will happen quite quickly.

Do you think it would make sense to talk about how to read the onchain data and how to withdraw the AppGateway fee balance?

Nowadays, I'm a fan of comments in sample code that build context and point to docs